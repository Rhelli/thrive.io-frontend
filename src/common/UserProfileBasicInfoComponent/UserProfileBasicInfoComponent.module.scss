.userProfileBasicContainer {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 20px;
  padding-top: 40px;
  background-color: #FDE2E6;
  border-bottom: 4px solid #FAA9B4;
}

.userProfileName {
  font-size: 24px;
  margin: 20px 10px 14px 10px;
}

.userProfileOccupationType {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-evenly;
  align-items: center;
  font-size: 18px;

  .userProfileBasicMidSpan {
    margin: 0 5px;
  }
}