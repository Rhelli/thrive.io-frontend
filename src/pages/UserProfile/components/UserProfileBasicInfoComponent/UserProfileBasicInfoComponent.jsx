import React from 'react';
import PropTypes from 'prop-types';
import styles from './UserProfileBasicInfoComponent.module.scss';

const UserProfileBasicInfoComponent = ({ userProfile }) => {
  const { name, occupation, userType } = userProfile;

  return (
    <div className={styles.userProfileBasicContainer}>
      <div className={styles.userProfileName}>
        <h2>{name}</h2>
      </div>
      <div className={styles.userProfileOccupationType}>
        {
          occupation ? (
            <span>
              <p>{occupation}</p>
            </span>
          ) : (
            <span>
              <p>No Occupation Yet</p>
            </span>
          )
        }
        <span className={styles.userProfileBasicMidSpan}>
          <p>|</p>
        </span>
        <span>
          <p>{userType}</p>
        </span>
      </div>
    </div>
  );
};

UserProfileBasicInfoComponent.propTypes = {
  userProfile: PropTypes.shape({
    name: PropTypes.string.isRequired,
    occupation: PropTypes.string,
    userType: PropTypes.string.isRequired,
  }).isRequired,
};

export default UserProfileBasicInfoComponent;
