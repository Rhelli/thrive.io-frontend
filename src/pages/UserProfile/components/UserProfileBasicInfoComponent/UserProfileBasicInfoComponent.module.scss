.userProfileBasicContainer {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 20px;
  margin-top: 40px;
  border: 1px solid purple;
}

.userProfileName {
  font-size: 20px;
  margin-bottom: 10px;
}

.userProfileOccupationType {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-evenly;
  align-items: center;

  .userProfileBasicMidSpan {
    margin: 0 5px;
  }
}