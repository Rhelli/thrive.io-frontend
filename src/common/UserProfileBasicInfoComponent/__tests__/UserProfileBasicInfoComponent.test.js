import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import UserProfileBasicInfoComponent from '../UserProfileBasicInfoComponent';

Enzyme.configure({ adapter: new Adapter() });

const userProfile = {
  name: 'Tonita',
  userType: 'Looking',
  occupation: 'Student',
};

const partialUserProfile = {
  name: 'Stevie G',
  userType: 'Looking',
  occupation: '',
};

const userProfileBasicInfoComponent = (
  <UserProfileBasicInfoComponent userProfile={userProfile} />
);

const partialUserProfileBasicInfoComponent = (
  <UserProfileBasicInfoComponent userProfile={partialUserProfile} />
);

describe('The appearance and functionality of the userProfileBasicInfo component', () => {
  const container = mount(userProfileBasicInfoComponent);
  const partialContainer = mount(partialUserProfileBasicInfoComponent);

  it('Matches the previous snapshot', () => {
    const tree = renderer.create(userProfileBasicInfoComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has the correct outer container with the correct attributes', () => {
    expect(container.childAt(0).hasClass('userProfileBasicContainer')).toBe(true);
  });

  it('Displays the users name correctly', () => {
    expect(container.find('.userProfileName').childAt(0).text()).toBe('Tonita');
  });

  it('Displays the users occupation when provided', () => {
    expect(container.find('.userProfileOccupationType').childAt(0).text()).toBe('Student');
  });

  it('Displays a placeholder if there is no occupation information available', () => {
    expect(partialContainer.find('.userProfileOccupationType').childAt(0).text()).toBe('No Occupation Yet');
  });

  it('Displays the user type correctly', () => {
    expect(container.find('.userProfileOccupationType').childAt(2).text()).toBe('Looking');
  });
});
