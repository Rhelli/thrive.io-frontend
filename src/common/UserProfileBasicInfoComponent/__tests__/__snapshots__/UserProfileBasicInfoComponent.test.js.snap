// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the userProfileBasicInfo component Matches the previous snapshot 1`] = `
<div
  className="userProfileBasicContainer"
>
  <div
    className="userProfileName"
  >
    <h2>
      Tonita
    </h2>
  </div>
  <div
    className="userProfileOccupationType"
  >
    <span>
      <p>
        Student
      </p>
    </span>
    <span
      className="userProfileBasicMidSpan"
    >
      <p>
        |
      </p>
    </span>
    <span>
      <p>
        Looking
      </p>
    </span>
  </div>
</div>
`;
